source/puppet/5.5/lang_data_abstract.md

```diff
@@ -117,6 +117,72 @@ Note that you can use capture groups in the regular expressions, but they won't
 * `Pattern[/\A[a-z].*/]` --- matches any string that begins with a lowercase letter.
 * `Pattern[/\A[a-z].*/, /\Anone\Z/]` --- matches the above **or** the exact string `"none"`.
 
+### `Timespan`
+
+The `Timespan` data type matches a Timespan; [strings][], or [numbers][] that represent a time and date; or `default`, which represents an infinite point in the past or future.
+
+It takes up to two parameters, and defaults to an infinite range to the past and future.
+
+#### Parameters
+
+The full signature for `Timespan` is:
+
+    Timespan[ (<TIMESPAN START OR LENGTH>, (<TIMESPAN END>)) ]
+
+Position | Parameter | Data Type | Default Value | Description
+---------|-----------|-----------|---------------|------------
+1 | Timespan Start or Length | String, Float, Integer, or `default` | `default` (-Infinity in a span) | Length of the timespan if passed alone, or the `from` value in a range if passed with a second parameter
+2 | Timespan End | String, Float, Integer, or `default` | `default` (+Infinity), or none if only one value is passed | The `to` value in a range
+
+`Timespan` values are interpreted depending on their format.
+
+* A String in the format of `D-HH:MM:SS` represents a span of days (`D`), hours (`HH`), minutes (`MM`), and seconds (`SS`)
+* An Integer or Float represents a number of seconds
+
+A `Timespan` range with either end defined as `default` (infinity) is an _open range_, while any other span is a _closed range_. The span's range is inclusive.
+
+The `Timespan` type is not enumerable.
+
```

Comment:
May want to link to the documentation for Timspan.new https://puppet.com/docs/puppet/6.0/function.html#conversion-to-timespan